iothub/service/src/DeliveryFailureReason.cs

```diff
@@ -1,15 +0,0 @@
-﻿// Copyright (c) Microsoft. All rights reserved.
```

Comment:
This enum was not being used at all. However, it is public so I wanted to bring it up to discussion here. It seems like a safe change and should not be breaking for customers but there is an off chance someone is simple using this enum though the SDK does not use it.

Reply:
Yes I've seen classes and enums with no references. But given it's publish, I always avoid deleting for the same reason you mentioned. Customers might be taking a dependency on this. Curious to know whether the folks who've been in the team longer how they handled such cases before.

Reply:
I'm all for removing defined types without references in the SDK, as long as we're sure it has no use.

There are a few that don't have references but do have uses. One I know of is in the import/export devices job because it defines the format of items in the file.

So be careful.

Reply:
Ok I will do some more analysis. Anyone else have any idea of this specific enum type?

Reply:
Are we throwing typed exception in these scenarios mentioned below? Good to check since we are removing if this was intended to be ToDo later.

Reply:
I don't know if we are throwing. It might be a TODO so it is out of scope for this PR. If anyone has more context on this, I can add an item on the board or we can take it forward. Otherwise if there is some place we would use it, I would like to know. I did not find any use of this yet.

Reply:
Yeah that's a good idea, we should add an item at least. Should also start conversation with service team what are the scenarios for Delivery Failures.

I wonder if this is related to status code feedback record status codes
`public FeedbackStatusCode StatusCode `

Reply:
Sure.